infrastructure/crypto/src/ristretto/pedersen.rs

```diff
@@ -92,13 +108,35 @@ impl<'a, 'b> Add for &'b PedersenOnRistretto255<'a> {
     }
 }
 
+/// Subtracts the left commitment from the right commitment
+/// #panics
+/// * If the base values are not equal
+impl<'b> Sub for &'b PedersenOnRistretto255 {
```

Comment:
I think I've figured out how to implement this only once on the parent (`HomomorphicCommitment`) type -- see #137 for similar approach on SchnorrSignature

Reply:
You still would need to define at least one version of the Add and one of the Sub implementations in this concrete implementation right? `HomomorphicCommitment` will still need to call down to the concrete Add and Sub at some point.

`HomomorphicCommitment` is going to see some refactoring based on this PR so I think it is best addressed there?